File: Scala/src/main/scala/Enemy/Enemy.scala
package Enemy

class Enemy(_lv: Int = 0, _hp: Int = 0, _defense: Int = 0, _attack: Int = 0, _hit: Int = 0, _flee: Int = 0, _name: String = "") {
  private var lv = _lv
  private var hp = _hp
  private var defense = _defense
  private var attack = _attack
  private var hit = _hit
  private var flee = _flee
  private var name = _name

  def Lv: Int = lv
  def Hp: Int = hp
  def Defense: Int = defense
  def Attack: Int = attack
  def Hit: Int = hit
  def Flee: Int = flee
  def Name: String = name

  def Lv_= (delta: Int): Unit = {
    lv = delta
  }
  def Hp_= (delta: Int): Unit = {
    hp = delta
  }
  def Defense_= (delta: Int): Unit = {
    defense = delta
  }
  def Attack_= (delta: Int): Unit = {
    attack = delta
  }
  def Hit_= (delta: Int): Unit = {
    hit = delta
  }
  def Flee_= (delta: Int): Unit = {
    flee = delta
  }
  def Name_= (delta: String): Unit = {
    name = delta
  }
}

File: SwiftProject/SwiftProject/Enemy.swift
//
//  Enemy.swift
//  SwiftProject
//
//  Created by 周哲輝 on 2019/06/24.
//  Copyright © 2019 周哲輝. All rights reserved.
//

import Foundation

public struct Enemy {
    let Lv: Int
    var Hp: Int
    let Defense: Int
    let Attack: Int
    let Hit: Int
    let Flee: Int
    let Name: String
    
    init() {
        Lv = 0
        Hp = 0
        Defense = 0
        Attack = 0
        Hit = 0
        Flee = 0
        Name = ""
    }
    
    init(lv: Int, hp: Int, defense: Int, attack: Int, hit: Int, flee: Int, name: String) {
        Lv = lv
        Hp = hp
        Defense = defense
        Attack = attack
        Hit = hit
        Flee = flee
        Name = name
    }
}
